<html>
<head>
<title>ErrorException</title>
<link rel="stylesheet" type="text/css" href="cppdoc.css" />
<link rel="shortcut icon" href="images/favicon.png">
</head>
<body>
<h1>The <code><a href=".">SimpleCxxLib</a></code> package</h1>
<hr>
<div class=include>#include "error.h"</div>
<h1 class=header><code>class ErrorException</code></h1>
This exception is thrown by calls to the <code>error</code>
function.  Typical code for catching errors looks like this:

<pre>
   try {
      ... code in which an error might occur ...
   } catch (ErrorException & ex) {
      ... code to handle the error condition ...
   }
</pre>

If an <code>ErrorException</code> is thrown at any point in the
range of the <code>try</code> (including in functions called from
that code), control will jump immediately to the error handler.
<table class=index width=100%>
<tr><td class=indexHead colspan=2>Function</td></tr>
<tr><td class=indexKey><nobr><a href="#Function:error">error(msg)</a>&nbsp;</nobr></td><td class=indexSynopsis width=100%>Signals an error condition in a program by throwing an <code>ErrorException</code> with the specified message.</td></tr>
</table>
<h2>Function detail</h2>
<hr>
<a name="Function:error"></a>
<pre class=detailCode>
void error(string msg);
</pre>
<div class=detailHTML>
Signals an error condition in a program by throwing an
<code>ErrorException</code> with the specified message.
<p>Usage:<br>
</div>
<pre class=usageCode>
error(msg);
</pre>
<hr>
</table>
</body>
</html>
